File: number-line-jumps/number-line-jumps.ts

const kangaroo = (x1: number, v1: number, x2: number, v2: number) => {
    if (x1 < x2 && v1 < v2) {
        return "NO"
    }
    else {
        if (v1 !== v2 && (x2 - x1)%(v2 - v1) === 0) {
            return "YES"
        } else {
            return "NO"
        }
    }
}

File: number-line-jumps/number-line-jumps.go
package main

func kangaroo(x1, v1, x2, v2 int) string {
	if (x1 < x2 && v1 <= v2) || (x2 < x1 && v2 <= v1) {
		return "NO"
	}
	// Check if the kangaroos meet after the same number of jumps
	if (x2-x1)%(v1-v2) == 0 {
		return "YES"
	}
	return "NO"
}